Assets/Scripts/Enemies/BossBehaviorController.cs

```diff
@@ -4,108 +4,156 @@
-        foreach(PhaseAction p in bossPhases[currentPhase].PossibleActions) {
-            p.behavior.StopAllCoroutines(); 
+    public void ChangePhase()
+    {
+        if (bossPhases[currentPhase].endBehaviour != null) bossPhases[currentPhase].endBehaviour.enabled = true;
+        foreach (PhaseAction p in bossPhases[currentPhase].backgroundActions) p.behavior.enabled = false;
+        foreach (PhaseAction p in bossPhases[currentPhase].PossibleActions)
+        {
+            p.behavior.StopAllCoroutines();
             p.behavior.enabled = false;
         }
 
         // if there are phases left
-        if(currentPhase<bossPhases.Count-1) {
+        if (currentPhase < bossPhases.Count - 1)
+        {
             currentPhase++;
-            StartPhase();
+            StartCoroutine(StartPhase());
         }
     }
 
-    public void StartPhase () {
+    public IEnumerator StartPhase()
+    {
         phaseHealth = bossPhases[currentPhase].health;
-        if( bossPhases[currentPhase].startBehaviour != null) bossPhases[currentPhase].startBehaviour.enabled = true;
-        foreach(PhaseAction p in bossPhases[currentPhase].backgroundActions) StartCoroutine( startActionBackground(p) );
-        ChangeAction();
+        if (bossPhases[currentPhase].startBehaviour != null)
+        {
+            bossPhases[currentPhase].startBehaviour.enabled = true;
+            while (bossPhases[currentPhase].startBehaviour.actionRunning)
+            {
+                yield return new WaitForEndOfFrame();
+            }
+            bossPhases[currentPhase].startBehaviour.enabled = false;
+        }
+        else ChangeAction();
+        foreach (PhaseAction p in bossPhases[currentPhase].backgroundActions) StartCoroutine(startActionBackground(p));
+
     }
-    public void ChangeAction () {
-        if(currentAction.behavior != null) currentAction.behavior.enabled = false;
-        if(actionQueue.Any()) { currentAction =  actionQueue[0]; actionQueue.RemoveAt(0); }
-        else if(bossPhases[currentPhase].PossibleActions.Count > 1) 
-            currentAction = bossPhases[currentPhase].PossibleActions.Where(x=> !x.Equals(currentAction)).ElementAt( rand.Next(bossPhases[currentPhase].PossibleActions.Count-1) );
+    public void ChangeAction()
+    {
+        foreach(PhaseAction p in actionQueue) print(p.behavior);
+        if (currentAction.behavior != null) currentAction.behavior.enabled = false;
+        if (bossPhases[currentPhase].repeatingBehaviour != null && currentAction.behavior != bossPhases[currentPhase].repeatingBehaviour) currentAction.behavior = bossPhases[currentPhase].repeatingBehaviour;
+        else if (actionQueue.Any()) { currentAction = actionQueue[0]; actionQueue.RemoveAt(0); }
+        else if (bossPhases[currentPhase].PossibleActions.Count > 1)
+            currentAction = bossPhases[currentPhase].PossibleActions.Where(x => !x.Equals(currentAction)).ElementAt(rand.Next(bossPhases[currentPhase].PossibleActions.Count));
```

Comment:
Need to change this to evaluate with the change provided in the actions 
IE when the manager is started the numbers are statically evaluated to their percentages and replaced (so that the chance of all actions adds to 1) and then that number can be used probably, need to think about the math efficenty for selecting randomly from a list